helm/templates/certificate_webhook-conf.yaml

```diff
@@ -1,10 +1,32 @@
 {{- $k8sMinor := (include "k8s-version-minor" .) -}}
+# Default to using a self-signed certificate
 {{- $altNames := list -}}
 {{- $altNames = append $altNames (printf "%s-svc" .Chart.Name) -}}
 {{- $altNames = append $altNames (printf "%s-svc.%s" .Chart.Name .Release.Namespace) -}}
 {{- $altNames = append $altNames (printf "%s-svc.%s.svc" .Chart.Name .Release.Namespace) -}}
 {{- $altNames = append $altNames (printf "%s-svc.%s.svc.cluster.local" .Chart.Name .Release.Namespace) -}}
-{{- $certs := genSelfSignedCert (printf "%s-svc.%s.svc" .Chart.Name .Release.Namespace) nil $altNames 36500 -}}
+{{- $certByHelmConfig := genSelfSignedCert (printf "%s-svc.%s.svc" .Chart.Name .Release.Namespace) nil $altNames 36500 -}}
+# Validate that if 'tls' key is set, either both or none of 'cert' and 'key' are set
+{{- if hasKey .Values.deployment "tls" -}}
+  {{- if and (not (hasKey .Values.deployment.tls "cert")) (hasKey .Values.deployment.tls "key")}}
+    {{ fail "Helm configuration has a 'deployment.tls' section with a 'key' attribute, but is missing the 'cert' attribute." -}}
+  {{- end -}}
+  {{- if and (not (hasKey .Values.deployment.tls "key")) (hasKey .Values.deployment.tls "cert")}}
+    {{ fail "Helm configuration has a 'deployment.tls' section with a 'cert' attribute, but is missing the 'key' attribute." -}}
+  {{- end -}}
+{{- end -}}
+# If present, use the configured certificate instead
+{{- if hasKey .Values.deployment "tls" -}}
```

Comment:
second check check for visibility? if not you can pull it in the condition clause above